tests/run-pass/panic/concurrent-panic.rs

```diff
@@ -0,0 +1,80 @@
+// ignore-windows: Concurrency on Windows is not supported yet.
+use std::sync::{Arc, Condvar, Mutex};
+use std::thread::{spawn, JoinHandle};
+
+struct BlockOnDrop(Option<JoinHandle<()>>);
+
+impl BlockOnDrop {
+    fn new(handle: JoinHandle<()>) -> BlockOnDrop {
+        BlockOnDrop(Some(handle))
+    }
+}
+
+impl Drop for BlockOnDrop {
+    fn drop(&mut self) {
+        let _ = self.0.take().unwrap().join();
+    }
+}
+
+/// Cause a panic in one thread while another thread is unwinding.
```

Comment:
Could you move this to the top of the file, and also say explicitly that this tests that we have per-thread panicking state?